packages/server-wallet/src/__test-with-peers__/utils.ts

```diff
@@ -35,7 +35,6 @@ export function waitForObjectiveEvent(
         handledObjectiveIds.add(o.objectiveId);
 
         if (handledObjectiveIds.size === _.uniq(objectiveIds).length) {
-          engine.removeListener(objectiveEventType);
```

Comment:
This was causing issues since it would remove **all** listeners for the event type. 

Since this is used only in tests and `engine.destroy` removes all listeners, I think it's fine to not remove it here.